services/QuillLMS/app/controllers/teachers/progress_reports/diagnostic_reports_controller.rb

```diff
@@ -192,6 +192,8 @@ def assign_post_test
   def assign_diagnostics
     params[:diagnostics].each do |diagnostic|
       Units::AssignmentHelpers.assign_unit_to_one_class(params[:classroom_id], diagnostic[:unit_template_id], diagnostic[:student_ids], false)
+      diagnostic_pack = UnitTemplate.find_by_id(diagnostic[:unit_template_id])
```

Comment:
This is a SUPER minor performance tweak that, given the size of the data involved, almost certainly won't matter, but since we only actually care about the `name` of the `UnitTemplate`, we can marginally reduce the size of the data payload that we fetch from the database by using `select`
```ruby
diagnostic_pack = UnitTemplate.select(:name).find_by_id(diagnostic[:unit_template_id])
```
This will create a `UnitTemplate` object that only has a `name` attribute.  Trying to call `diagnostic_pack.id` will error in this case because we didn't fetch that data.  Again, in this case, it's only one row, so we're not really saving much by fetching only the name, but I figured I'd mention it since there are use-cases where this can matter.